src/node_crypto.cc

```diff
@@ -3261,7 +3261,7 @@ void CipherBase::InitIv(const char* cipher_type,
      iv_length (0) for ECB ciphers */
   if (EVP_CIPHER_iv_length(cipher_) != iv_len &&
       !(EVP_CIPHER_mode(cipher_) == EVP_CIPH_ECB_MODE && iv_len == 0) &&
-      !(EVP_CIPHER_mode(cipher_) == EVP_CIPH_GCM_MODE) && iv_len > 0) {
+      !(EVP_CIPHER_mode(cipher_) == EVP_CIPH_GCM_MODE && iv_len > 0)) {
     return env()->ThrowError("Invalid IV length");
```

Comment:
I could not figure out the reason why the GCM test of zero length of iv_len in `test-crypto-authenticated.js` has passed.  There seems two reasons. One is the test has a bug to skip some tests due to checking `test.password` and `i>0` but iv_len in GCM can be checked in the following `EVP_CIPHER_CTX_ctrl(&ctx_, EVP_CTRL_GCM_SET_IVLEN, iv_len, nullptr)`. 
I think the fix of  `test-crypto-authenticated.js` as in https://gist.github.com/shigeki/afc23ecef12006777b668a2b8a24b044 is also needed.